packages/http-client-java/emitter/src/validate.ts

```diff
@@ -0,0 +1,71 @@
+import { Program } from "@typespec/compiler";
+import { logError, spawnAsync } from "./utils.js";
+
+export const JDK_NOT_FOUND_MESSAGE =
+  "Java Development Kit (JDK) is not found in PATH. Please install JDK 17 or above. Microsoft Build of OpenJDK can be downloaded from https://learn.microsoft.com/java/openjdk/download";
+
+export async function validateDependencies(
+  program: Program | undefined,
+  logDiagnostic: boolean = false,
+) {
```

Comment:
I didn't update the signature here, as original this is intended to be called in postinstall script as well (where there is no `program`).